deltacat/compute/compactor/compaction_session.py

```diff
@@ -430,6 +440,15 @@ def _execute_compaction_round(
     mat_results = sorted(mat_results, key=lambda m: m.task_index)
     deltas = [m.delta for m in mat_results]
     merged_delta = Delta.merge_deltas(deltas)
+    assert (
+        total_hb_record_count - total_dd_record_count == merged_delta.meta.record_count
+    ), (
```

Comment:
Can we also audit this info? It's worth looking at this even when assertion doesn't fail to understand what's going on.

Reply:
Sure, by auditing are we talking about simply logging here or a dump of this info into some data store?

Reply:
Deferring this to a separate PR/issue #127

Reply:
sure. Can we atleast log it irrespective of assertion result? Here, it's getting logged only if the assertion fails.